### PR TITLE
[docs] Fix imports for x-grid-data-generator

### PR DESCRIPTION
**Pull request by DanailH**
In response to this comment https://github.com/mui-org/material-ui-x/pull/855#discussion_r560451601

I'm not sure if the `x-data-grid-generator` is supposed to work with just the `data-grid` module. Let's discuss it here. Also if it should work then the name would also need to change IMO (remove the `x-` prefix).

### CODE REVIEW

**Review by oliviertassinari — CHANGES_REQUESTED** [review]
OK, so we are going with option 1. sounds fair.

> Also if it should work then the name would also need to change IMO (remove the x- prefix).

Regarding the naming, if we were to improve it, I would personally suggest that we create a new repository under this folder: `/packages/material-ui-x`. Then, we could do `import {} from '@material-ui/x/dataGenerator';`

**Comment by DanailH** [comment]
I understand that it is still broken. I opened the PR to use it as a discussion to decide between option 1 and option 2.

**Comment by DanailH** [comment]
After giving it a bit more thought, removing `data-grid` from the dependencies looks like the way to go. I had to duplicate the v5 helpers which can be removed once we decide to fully migrate so that code is only temporary. The cell definition is needed.

**Comment by oliviertassinari** [comment]
What's the pro of option 2 compared to option 1? I mean, what's the selling point?

**Comment by DanailH** [comment]
Removing DataGrid from the dep would make using this simpler as you can use it with both right. Correct me if I'm wrong but if we keep it as it is and want to make it work with `x-grid` then you would need `data-grid` installed as well right, how else could you know from where to import the `CellParams` for example?

**Comment by oliviertassinari** [comment]
@DanailH OK, I was interested in the rationale. We trade of bit of code duplication for one less dependency in the codesandbox (when using x-grid). No objections, we can give it a try.

**Comment by dtassone** [comment]
We could move it to the grid folder as well, and build it the same way we build data-grid and x-grid package. 
So we remove the dependency on DataGrid and we don't duplicate code

**Comment by DanailH** [comment]
Ok, I moved the `x-grid-data-generator` under `packages/grid` and fixed the build (made it the same way it is done for the `data-grid` and `x-grid`)

**Review by oliviertassinari — APPROVED** [review]
Looks great

File: packages/grid/tsconfig.build.json [hunk 1]
```diff
@@ -9,6 +9,6 @@
     "types": ["jest", "node"],
     "rootDir": "./"
   },
-  "include": ["./data-grid/src", "./x-grid/src", "./_modules_/**/*"],
+  "include": ["./data-grid/src", "./x-grid/src", "./x-grid-data-generator/src", "./_modules_/**/*"],
```
Comment (dtassone) [inline packages/grid/tsconfig.build.json:12]: `/x-grid-data-generator/src` don't we want to rename this folder?

Comment (DanailH) [reply packages/grid/tsconfig.build.json:12]: More appropriate would be 'grid-data-generator' but then there would be a mismatch between the name of the package and the name of the folder.

Comment (DanailH) [reply packages/grid/tsconfig.build.json:12]: For now, we will keep it as it is and review it at a later stage.

Comment (oliviertassinari) [reply packages/grid/tsconfig.build.json:12]: The related RFC: https://github.com/mui-org/material-ui-x/issues/226.

**Pull request merged**